src/test/java/org/apache/datasketches/quantiles/DirectQuantilesMemoryRequestTest.java

```diff
@@ -47,35 +46,30 @@ public void checkLimitedMemoryScenarios() { //Requesting application
     final int initBytes = ((2 * k) + 4) << 3; //just the BB
 
     //########## Owning Implementation
-    // This part would actually be part of the Memory owning implemention so it is faked here
-    try (WritableHandle wdh = WritableMemory.allocateDirect(initBytes,
-            ByteOrder.nativeOrder(), new DefaultMemoryRequestServer())) {
-      final WritableMemory wmem = wdh.getWritable();
-      println("Initial mem size: " + wmem.getCapacity());
-
-      //########## Receiving Application
-      // The receiving application has been given wmem to use for a sketch,
-      // but alas, it is not ultimately large enough.
-      final UpdateDoublesSketch usk1 = DoublesSketch.builder().setK(k).build(wmem);
-      assertTrue(usk1.isEmpty());
-
-      //Load the sketch
-      for (int i = 0; i < u; i++) {
-        // The sketch uses The MemoryRequest, acquired from wmem, to acquire more memory as
-        // needed, and requests via the MemoryRequest to free the old allocations.
-        usk1.update(i);
-      }
-      final double result = usk1.getQuantile(0.5);
-      println("Result: " + result);
-      assertEquals(result, u / 2.0, 0.05 * u); //Success
-
-      //########## Owning Implementation
-      //The actual Memory has been re-allocated several times,
-      // so the above wmem reference is invalid.
-      println("\nFinal mem size: " + wmem.getCapacity());
-    } catch (Exception e) {
-      throw new RuntimeException(e);
+    // This part would actually be part of the Memory owning implementation so it is faked here
+    WritableMemory wmem = WritableMemory.allocateDirect(initBytes, ByteOrder.nativeOrder(), new DefaultMemoryRequestServer());
+    println("Initial mem size: " + wmem.getCapacity());
+
+    //########## Receiving Application
+    // The receiving application has been given wmem to use for a sketch,
+    // but alas, it is not ultimately large enough.
+    final UpdateDoublesSketch usk1 = DoublesSketch.builder().setK(k).build(wmem);
+    assertTrue(usk1.isEmpty());
+
+    //Load the sketch
+    for (int i = 0; i < u; i++) {
+      // The sketch uses The MemoryRequest, acquired from wmem, to acquire more memory as
+      // needed, and requests via the MemoryRequest to free the old allocations.
+      usk1.update(i);
     }
+    final double result = usk1.getQuantile(0.5);
+    println("Result: " + result);
+    assertEquals(result, u / 2.0, 0.05 * u); //Success
+
+    //The actual Memory has been re-allocated several times,
+    // so the the wmem reference is invalid. Use the sketch to get the last memory reference.
+    WritableMemory lastMem = usk1.getMemory();
```

Comment:
The real objective of this test is to be an example for the user on how the MemoryRequestServer can be used to help with limited memory scenarios where the sketch needs to expand beyond what was allocated.  This is why there are extensive code comments in this test method.

The actual test that the expansion of memory occurred (several times) and the sketch continued successfully is on line 67.   After that, the actual capacity of the last memory allocated is printed out just for the user.  There is no reason to test it, we know that it was big enough.

Reply:
I think in order to call it a test and pretend it means anything we should assert that the size is larger than the initial size.

Reply:
The test of success is on line 67.  By definition, if line 67 passes, the memory had to grow larger than the initial size.  If the memory capacity did not grow larger than the initial size, the sketch would run out of memory and the test would fail.  In fact it would fail before it ever reached line 67.  So putting an extra test that the memory in fact grew beyond the initial size is redundant and impotent because that extra test would either never be executed or it would always pass.